spek-runner/junit5/src/main/kotlin/org/spekframework/spek2/junit/RuntimeExecutionListenerAdapter.kt

```diff
@@ -23,7 +22,12 @@ class RuntimeExecutionListenerAdapter(val listener: EngineExecutionListener,
     override fun executionFinish() { }
 
     override fun testExecutionStart(test: TestScopeImpl) {
-        listener.executionStarted(test.asDescriptor())
+        val descriptor = test.asDescriptor()
+        if (test.parent is ActionScopeImpl) {
+            listener.dynamicTestRegistered(descriptor)
```

Comment:
Does it handle this kind of situation?

```kotlin
listOf(1,  2, 3).forEach { number ->

    context("number is [$number]") {
    
        beforeEachTest {
            println(number)
        }
    }
}
```
I’m asking mostly because of `ActionScopeImpl`.

Reply:
That should work, but I'm not sure how it relates to the PR?

Reply:
I’m just not seeing the `ActionScopeImpl` check in the previous version of the code.

Reply:
Ahh, your example should still work. Only affects `test`s whose parent is an `action`, which `context` is not.